src/Host/Protocol/Test/UserProfileServicePipe/UserProfileServicePipeTest.cs

```diff
@@ -0,0 +1,135 @@
+                        f.ShouldNotThrow();
+                    } else {
+                        Func<Task> f = () => RUserProfileCreator.CreateProfileAsync(serverTimeOutms: TestServerTimeoutMS, clientTimeOutms: TestClientTimeoutMS, userProfileService: creator);
+                        await f.ShouldThrowAsync<Exception>();
+                    }
+                }
+                testDone.Set();
+            }).DoNotWait();
+
+            using (CancellationTokenSource cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(TestClientTimeoutMS))) {
+                UserProfileResultMock result = await CreateProfileClientTestWorkerAsync(input, cts.Token);
+                if (isFuzzTest) {
+                    // fuzz test parsing succeeded, the creator always fails for this test.
+                    result?.Error.Should().Be(13);
+                } else {
+                    if (isValidParse) {
+                        result.Error.Should().Be((uint)(isValidAccount ? 0 : 13));
+                        result.ProfileExists.Should().Be(isExistingAccount);
+                    } else {
+                        result.Should().BeNull();
+                    }
+                }
+            }
+
+            testDone.Wait();
+        }
+        private int TestServerTimeoutMS => 5000;
+        private int TestClientTimeoutMS => 5000;
+
+        [CompositeTest]
+        // valid test data
+        [InlineData("{\"Username\":\"testname\", \"Domain\":\"testdomain\", \"Password\":\"testPassword\"}", "testname", "testdomain", "testPassword", true, true, false)]
+        // Missing quote
+        [InlineData("{Username\":\"testname\", \"Domain\":\"testdomain\", \"Password\":\"testPassword\"}", null, null, null, false, false, false)]
+        // Missing closing paren
+        [InlineData("{", null, null, null, false, false, false)]
+        // empty json object
+        [InlineData("{}", null, null, null, false, false, false)]
+        // No username, domain and password
+        [InlineData("{\"Username\":, \"Domain\":, \"Password\":}", null, null, null, false, false, false)]
+        // empty username domain password
+        [InlineData("{\"Username\": \"\", \"Domain\": \"\", \"Password\": \"\"}", "", "", "", true, false, false)]
+        // whitespace input string
+        [InlineData("                     ", null, null, null, true, false, false)]
+        public async Task CreateProfileTest(string input, string username, string domain, string password, bool isValidParse, bool isValidAccount, bool isExistingAccount) {
+            var creator = UserProfileCreatorMock.Create(username, domain, password, isValidParse, isValidAccount, isExistingAccount);
+            await CreateProfileTestRunnerAsync(creator, input, isValidParse, isValidAccount, isExistingAccount);
+        }
+
+        [Test]
```

Comment:
We should probably have a distinct category for fuzz tests. Take a look at how other test classes use test category attributes.